File: sdk/src/main/java/com/checkout/sdk/request/CardTokenisationRequest.java
package com.checkout.sdk.request;

import com.checkout.sdk.billingdetails.model.BillingDetails;

/**
 * The request model object for the card tokenisation request
 */
public class CardTokenisationRequest {

    private String number;
    private String name;
    private String expiryMonth;
    private String expiryYear;
    private String cvv;

    private BillingDetails billingDetails;

    public CardTokenisationRequest(String number, String name, String expiryMonth, String expiryYear, String cvv, BillingDetails billingDetails) {
        this.number = number;
        this.name = name;
        this.expiryMonth = expiryMonth;
        this.expiryYear = expiryYear;
        this.cvv = cvv;
        this.billingDetails = billingDetails;
    }

    public CardTokenisationRequest(String number, String name, String expiryMonth, String expiryYear, String cvv) {
        this.number = number;
        this.name = name;
        this.expiryMonth = expiryMonth;
        this.expiryYear = expiryYear;
        this.cvv = cvv;
    }
}

File: sdk/src/main/java/com/checkout/sdk/request/CardTokenisationRequest.kt
package com.checkout.sdk.request

import com.checkout.sdk.billingdetails.NetworkBillingModel

/**
 * The request model object for the card tokenisation request
 */
class CardTokenisationRequest(
    private val number: String,
    private val name: String,
    private val expiryMonth: String,
    private val expiryYear: String,
    private val cvv: String,
    private val billingModel: NetworkBillingModel?
)
